### ISSUE TITLE
Export Config type

### ISSUE THREAD

**Issue by connorjs** [stylelint/stylelint#6733]
### What is the problem you're trying to solve?

I prefer authoring configuration files in JavaScript, not JSON, for out-of-the-box comment support and general JavaScript language features (share via JS code).

As part of this, I want my IDE to assist me with auto-complete. _← The feature request._

### What solution would you like to see?

I would like stylelint to export its `Config` type ([permalink](https://github.com/stylelint/stylelint/blob/0bc53af31e85d76db9407801659bc850998042ad/types/stylelint/index.d.ts#L88-L109)). Then, I can use the following in my `stylelint.config.cjs` file.

```js
/** @type {import('stylelint').Config} */
module.exports = {
	// This will auto complete
}
```

I think all we need to do would be to add `export` here. I believe we do NOT need to add export to the "sub types" (auto-complete will see the resolved type and work).

**Comment by ybiquitous** [comment]
@connorjs Thanks for the proposal. Sounds good to me. 👍🏼 

@emcohaszing Do you have any thoughts?

**Comment by ybiquitous** [comment]
Oh, sorry for my bad mention. 😓 

@remcohaszing Do you have any good idea about this?

**Comment by remcohaszing** [comment]
If a namespace has an export, then all other members are private. If it has no explicit exports, then all members are exported.

The `stylelint` namespace has no explicit exports, so all members are already exported, including `Config`. You could explicitly export all members to have the same result. That's a purely stylistic choice, but IMO that's just extra noise. Currently all private members are kept outside of the namespace.

**Comment by ybiquitous** [comment]
@remcohaszing Thanks for the quick answer! I confirmed the `Config` type is exported:

For example, assuming `.stylelintrc.cjs` with the following content:

```js
/** @type {import('stylelint').Config} */
const config = {
  extendss: ["stylelint-config-standard"],
};

module.exports = config;
```

Type-checking fails:

```console
$ npx tsc
.stylelintrc.cjs:3:3 - error TS2322: Type '{ extendss: string[]; }' is not assignable to type 'Config'.
  Object literal may only specify known properties, but 'extendss' does not exist in type 'Config'. Did you mean to write 'extends'?

3   extendss: ["stylelint-config-standard"],
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in .stylelintrc.cjs:3
```

@connorjs What do you think?

**Comment by connorjs** [comment]
I will need to revisit my configuration then. My IDE does not see the `Config` type and does not emit warnings with the `extendss` typo. Thus, auto-complete is not working for me.

However, given you successfully reproduced `npc tsc` failing, I’m fine closing this as no issue.

**Comment by ybiquitous** [comment]
Thanks for the response. I'm closing this issue.

If you find some solution, sharing it might be helpful here for people who meet the same problem.
